qiskit/algorithms/observables_evaluator.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import annotations
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.opflow import PauliSumOp
+from . import AlgorithmError
+from .list_or_dict import ListOrDict
+from ..primitives import EstimatorResult, BaseEstimator
+from ..quantum_info.operators.base_operator import BaseOperator
+
+
+def estimate_observables(
+    estimator: BaseEstimator,
+    quantum_state: QuantumCircuit,
+    observables: ListOrDict[BaseOperator | PauliSumOp],
+    threshold: float = 1e-12,
+) -> ListOrDict[tuple[complex, tuple[complex | None, int | None]]]:
+    """
+    Accepts a sequence of operators and calculates their expectation values - means
+    and standard deviations. They are calculated with respect to a quantum state provided. A user
+    can optionally provide a threshold value which filters mean values falling below the threshold.
+
+    Args:
+        estimator: An estimator primitive used for calculations.
+        quantum_state: An unparametrized quantum circuit representing a quantum state that
+            expectation values are computed against.
+        observables: A list or a dictionary of operators whose expectation values are to be
+            calculated.
+        threshold: A threshold value that defines which mean values should be neglected (helpful for
+            ignoring numerical instabilities close to 0).
+
+    Returns:
+        A list or a dictionary of tuples (mean, standard deviation).
+
+    Raises:
+        ValueError: If a ``quantum_state`` with free parameters is provided.
+        AlgorithmError: If a primitive job is not successful.
+    """
+
+    if (
+        isinstance(quantum_state, QuantumCircuit)  # State cannot be parametrized
+        and len(quantum_state.parameters) > 0
+    ):
+        raise ValueError(
+            "A parametrized representation of a quantum_state was provided. It is not "
+            "allowed - it cannot have free parameters."
+        )
+    if isinstance(observables, dict):
+        observables_list = list(observables.values())
+    else:
+        observables_list = observables
+
+    observables_list = _handle_zero_ops(observables_list)
+    quantum_state = [quantum_state] * len(observables)
+    try:
+        estimator_job = estimator.run(quantum_state, observables_list)
+        expectation_values = estimator_job.result().values
+    except Exception as exc:
+        raise AlgorithmError("The primitive job failed!") from exc
+
+    variance_and_shots = _prep_variance_and_shots(estimator_job, len(expectation_values))
+
+    # Discard values below threshold
+    observables_means = expectation_values * (np.abs(expectation_values) > threshold)
+    # zip means and standard deviations into tuples
+    observables_results = list(zip(observables_means, variance_and_shots))
+
+    return _prepare_result(observables_results, observables)
+
+
+def _handle_zero_ops(
+    observables_list: list[BaseOperator | PauliSumOp | int],
```

Comment:
The int here seems weird. Since we can only pass at the top level at `BaseOperator | PauliSumOp` the notion of zero/empty operator, in Aqua we tested the operator with an is_empty function. Opflow was to do an == 0 against the operator. Never is an int passed its always a valid operator it may just be empty. Whether that needs special treatment or not against the Estimator - we said either way we would check here. I am not sure what BaseOperator equality does in regards of == 0